javalib/src/main/scala/java/net/URLDecoder.scala

```diff
@@ -0,0 +1,46 @@
+package java.net
+
+import java.io.UnsupportedEncodingException
+import java.nio.{CharBuffer, ByteBuffer}
+import java.nio.charset.Charset
+
+object URLDecoder {
+  def decode(s: String, enc: String): String = {
+
+    if (!Charset.isSupported(enc)) throw new UnsupportedEncodingException(enc)
```

Comment:
well it actually throws an UnsupportedCharsetException 
https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/nio/charset/Charset.scala#L62